File: euler/src/1 - Multiples of 3 or 5.ts
import { calcSum } from './utils';

const below = 1000;
const multiplesOf = [3, 5];
const toAdd: Set<number> = new Set();

// Find all of the multiples of 3 and 5 below 1000
for (let i = 1; i < below; i++) {
    multiplesOf.forEach(num => (i % num == 0 ? toAdd.add(i) : null));
}

console.log(calcSum(Array.from(toAdd)));

File: euler/_pythonarchive/1 - Multiples of 3 or 5.py
below = 1000
multiplesOf = [3,5]
toAdd = []

for i in range(1, below):
	for num in multiplesOf:
		if i % num == 0:
			toAdd.append(i)

toAdd = list(dict.fromkeys(toAdd))
total = 0

for i in toAdd:
	total += i

print(total)
